onnxruntime/core/providers/nnapi/nnapi_builtin/builders/shaper.cc

```diff
@@ -371,25 +371,29 @@ void Shaper::AddShape(const std::string& name, const Shape& shape) {
   shape_map_[name] = shape;
 }
 
-void Shaper::UpdateShape(const std::string& name, const Shape& new_shape) {
-  ORT_ENFORCE(shaper_finalized_,
-              "Cannot UpdateShape while shaper is not finalized");
+Status Shaper::UpdateShape(const std::string& name, const Shape& new_shape) {
+  ORT_RETURN_IF_NOT(shaper_finalized_,
+                    "Cannot UpdateShape while shaper is not finalized");
```

Comment:
I'm hoping we can just get rid of shaper_finalized_ given this class has internal usage only. If not, why do we return a failure here, but somewhat silently ignore it in earlier code (e.g. we do a bunch of work and don't add things to shape_ops_ if set in earlier methods)?